## np.random.randn(d0, d1, ..., dn)
https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randn.html

n차원 정수를 입력해주면

그 차원에 맞는 "standard normal" distribution 랜덤값들을 출력해준다 = 표준정규분포

표준정규분포는 평균이 0 이고 표준편차가 1이다
https://www.oswego.edu/~srp/stats/z.htm

표준편차가 뭔지 애매할텐데 볼록한 종모양의 분포 그래프에서
- 평균 앞뒤로 표준편차만큼의 공간에 약 68%가 있고
- 평균 앞뒤로 표준편차 * 2 만큼의 공간에 약 95%
- 평균 앞뒤로 표준편차 * 3 만큼의 공간에 약 99.7% 가 있다는 것이다


In [2]:
import numpy as np
W = np.random.randn(5,5)
print(W)

[[ 0.67196524 -1.24712706  1.17262306  0.176544   -1.39476247]
 [ 2.11293365 -0.55523176  0.30976635  1.15910232  2.22269391]
 [-0.64284049 -0.91189978 -0.9253139  -1.10520488 -0.62402671]
 [-1.00473896  0.28278934  1.44218647 -0.27708792  0.87141903]
 [-0.3385773   0.66252922 -0.35962214 -1.08943987 -0.44004925]]


위 코드를 돌리니 아래와 같이 나왔다.

[[ 0.67196524 -1.24712706  1.17262306  0.176544   -1.39476247]

 [ 2.11293365 -0.55523176  0.30976635  1.15910232  2.22269391]
 
 [-0.64284049 -0.91189978 -0.9253139  -1.10520488 -0.62402671]
 
 [-1.00473896  0.28278934  1.44218647 -0.27708792  0.87141903]
 
 [-0.3385773   0.66252922 -0.35962214 -1.08943987 -0.44004925]]
 
 
 대략 Standard Normal Distribution 대로 이다. 
 * -1에서 1 사이가 68%
 * -2에서 2 사이가 95%
 * -3에서 3 사이가 99.7 프로

## numpy.bincount(myArray)

array에 있는 같은 값들의 갯수

## numpy.argmax(counts)

가장 큰 값의 index를 리턴

In [6]:
import numpy as np
a = np.array([1,2,3,1,2,1,1,1,3,2,2,1,4,4,4,4,4,4,4,4,4])
counts = np.bincount(a)

# 0은 0개, 1은 6개, 2는 4개, 3은 2개, 4는 9개이다.
# 따라서 (어찌보면 count list의 index와 같은 정수의 갯수를 의미하는 list인) counts = [0 6 4 2 9] 가 된다
print(counts)  

# numpy.argmax는 list에서 가장 큰 수의 인덱스를 리턴해준다
print (np.argmax(counts))

# numpy.argmax() 예시 하나더
mycount = [3, 5, 6, 2, 1, 7]
print(np.argmax(mycount))  # 가장 큰 수인 7의 index인 5를 리턴해준다

[0 6 4 2 9]
4
5


## numpy.argsort()

리스트를 넣어주면
리스트에서 작은 값들 순서로 값들의 인덱스를 리턴해준다



In [8]:
print(np.argsort([5, 9, 3, 4, 2, 7]))  # 작은 값들의 인덱스부터 리턴해줌. 
# 즉, 2가 가장 작으니깐 2의 인덱스인 4
# 그다음은 3이니깐 인덱스인 2
# 4는 3, 5는 0, 7은 5, 9는 1이 된다. 따라서 [4,2,3,0,5,1]이 리턴된다

[4 2 3 0 5 1]
